livekit-plugins/livekit-plugins-cartesia/livekit/plugins/cartesia/tts.py

```diff
@@ -319,14 +315,11 @@ def _send_last_frame(*, segment_id: str, is_final: bool) -> None:
                 if data.get("data"):
                     b64data = base64.b64decode(data["data"])
                     for frame in audio_bstream.write(b64data):
-                        _send_last_frame(segment_id=segment_id, is_final=False)
-                        last_frame = frame
+                        _send_audio_frame(frame, segment_id, False)
                 elif data.get("done"):
-                    for frame in audio_bstream.flush():
-                        _send_last_frame(segment_id=segment_id, is_final=False)
-                        last_frame = frame
-
-                    _send_last_frame(segment_id=segment_id, is_final=True)
+                    frames = list(audio_bstream.flush())
+                    for i, frame in enumerate(frames):
+                        _send_audio_frame(frame, segment_id, i == len(frames) - 1)
```

Comment:
This doesn't guarantee that at least one frame with is_final=True is sent.
Because `audio_bstream.flush()` could return an empty list in some scenarios

Reply:
ah you're right